### DISCUSSION TITLE
DuckDBPyRelation interface is not complete?

### DISCUSSION THREAD

**Discussion by goodwanghan** [duckdb/duckdb#2893] [Q&A]
Hello,

I think `DuckDBPyRelation` is a very good idea. But I notice the programming interface lacks some important features, for example:

1. `join` can't specify join type
2. `union`, `intersect`, `except_` can't specify distinct or not
3. `write_csv` doesn't have any option.

For 1, 2 I have to create views from relations and then use standard SQL for the operations. I wonder if creating a view will have any time and space overhead? Does it support lazy evaluation? After use, do we need to clean up the views to save space?

Thanks!

**Comment by Alex-Monahan** [comment]
Please have a look at our Relational API TODO list here! Please add in what you are missing to that list by adding a comment on that issue:
#2000

**Reply by goodwanghan** [reply]
Oh I think `DuckDBPyRelation` does not follow pandas (which I think is a good direction). But I will add the requests there. Thanks!

**Comment by Alex-Monahan** [comment]
Views don't take up space and the API is lazily evaluated!

**Reply by goodwanghan** [reply]
Thank you!
